lib/Myriad/RPC/Client/Implementation/Memory.pm

```diff
@@ -88,14 +88,21 @@ async method call_rpc ($service, $method, %args) {
             $self->loop->timeout_future(at => $deadline),
             $pending
         );
+
+        unless (exists $message->response->{message}) {
+            my $reason = $message->response->{error}{message} // "Unknown";
+            Myriad::Exception::RPC::RemoteException->throw(reason => "Remote exception is thrown: $reason")
```

Comment:
The logic here doesn't look right - we're checking for `message` and `response`, why split up the conditional like this?

Reply:
Could you please clarify on "split up the conditional"?